enhancements/coreos-encrypted-by-default.md

```diff
@@ -0,0 +1,425 @@
+  * Randomize the LUKS container's UUID,
+  * Re-encrypt the LUKS container using AES-256-CBC (FIPS 140-2 compliant). This process generates a new random master key.
+  * If a Clevis configration is found, apply it in the initramfs. Otherwise:
+      * if a TPM2 device is found, bind the LUKS container to TPM2.
+      * if no TPM2 device, use the plaintext Clevis pin
+
+
+The Dracut modules will also allow both first-boot and reboot re-encryption using kernel commandline arguments. The re-encryption step is necessary to ensure that the LUKS master key is unique to each instance of CoreOS. In testing, the re-encryption step takes between 1 and 3 minutes, depending on the CPU and the backing disk-type.
+
+Afer first-boot, a Linux systemd service will provide for re-applying a Clevis configuration. If the service finds a configuration, it will:
+  * apply the Clevis configuration
+  * write a new random passphrase in /root
+  * revoke the prior Clevis configuration
+
+The service will be activated when `/etc/clevis.json` is found, alloing for a Clevis profile provided installation time, or through a Machine Config. Once a Clevis configuration has is applied, the LUKS container will be automatically opened at boot if the conditions of the configuration allow so.  Otherwise, to facilitate unattended boot, then a plaintext Clevis configuration will be used.
+
+### User Stories
+
+#### Story 1: Requirements
+
+ACME Corp is only able to consider technologies that have encrypted disks, but they want to use Google Cloud. With Openshift's 4.3 boot images, they can evaluate Openshift.
+
+#### Story 2: Compliance
+
+ACME Corp deploys a new Openshift cluster to Microsoft Azure. During an audit, the compliance officer notes that data should be encrypted at rest with a key-escrow in Cloud KMS. During lunch, the IT department deploys a Machine Config with a Clevis configuration, which CoreOS actions. After lunch, the IT department reports that the cluster is compliant with the policy.
+
+#### Story 3: Upgrades
+
+ACME Corp has a 4.1 cluster and is upgrading to 4.3 and wants to use the new encryption. The IT department simply:
+* Adds their Clevis machine config set
+* [updates the machine set](https://docs.openshift.com/container-platform/4.1/machine_management/creating-machineset.html)
+* Scales up new nodes
+* Scales down the old nodes
+
+#### Story 4: Edge Clusters
+
+ACME Corp deploys sensitive bare-metal clusters to the edge. They build their clusters with a Clevis profile using TPM2 and Tang. Evil Corp clandestinely steals a few of ACME Corp's nodes. Since ACME Corp's nodes are configured using NBE encryption, no-data is compromised. ACME Corp avoids being shamed on the evening news and withstands the ire of the Board of Directors.
+
+### Implementation Details/Notes/Constraints [optional]
+
+This proposal introduces dependencies on CoreOS only; it is an OS-level implementation. Openshift itself will entirely unaware that CoreOS is encrypted.
```

Comment:
BTW I don't see such a hard line between "OpenShift" and "CoreOS" in this context - the MCO specifically bridges them.  And I think it'd indeed be nice down the line if e.g. one could tweak a `MachineConfig` option to say change encryption algorithms (or switch to using a Tang server) - and the MCO would know how to reprovision workers with a new Ignition config that would accomplish that.